units/host_setup_script.go

```diff
@@ -116,6 +118,21 @@ func (j *hostSetupScriptJob) Run(ctx context.Context) {
 			j.AddRetryableError(errors.Wrap(err, "checking if task data is fetched yet"))
 			return
 		}
+		// once task data is fetched, we no longer need the github token. Revoke it here
+		// in case something went wrong and it's still around.
+		if f := j.host.ProvisionOptions.FetchOpts; f != nil {
+			_ = thirdparty.RevokeInstallationToken(ctx, f.GithubAppToken)
```

Comment:
GetInstallationToken uses a cached token, so we shouldn't revoke I think. cc @Kimchelly can you check this?

Reply:
That's correct, as per the doc comment on `RevokeInstallationToken`, it's invalid to call it with a cached token. It's also the case that this will only revoke if the user chooses to run a custom setup script (which is pretty uncommon). If they don't set a custom setup script, this job never runs, so it'll never revoke the token.

If this is being given to a user's spawn host, it probably shouldn't use a cached token. The token caching is for make internal Evergreen usage more efficient, not for users.

Reply:
This PR is from two weeks ago so it predates the work to start caching Installation tokens. I'll read it again to make sure it still works as expected. 
This is using a more scoped down read only token, so I don't think it makes sense to use the cache, as those are full-permission tokens.